cEP-0020.md

```diff
@@ -0,0 +1,462 @@
+Badges and Levels. For each of these elements, we will have a set of rules to
+describe the operation of the gamified environment. i.e.:
+
+### Activities
+
+Let's start by defining some activities which could be tracked:
+
+- Create a bug report
+- Create newcomer difficulty issue
+- Create a bear proposal issue
+- Create a documentation issue
+- Create low difficulty issue
+- Create medium difficulty issue
+- Create pull request that fixes a bug
+- Create pull request for bear proposal
+- Create pull request for a documentation issue
+- Create pull request for newcomer issue
+- Create pull request for low difficulty issue
+- Create pull request for medium difficulty issue
+- Gitmate errors on pull request
+- CI status of the pull request
+- Review comments on pull request
+- Number of questions asked in `coala/coala` room
+- Number of answers given in `coala/coala` room
+- Number of questions asked in `coala-bears` room
+- Number of  answers given in `coala-bears` room
+- Number of questions asked in `coala/corobo` room
+- Number of answers given in `coala/corobo` room
+- Number of questions asked in `coala/community` room
+- Number of answers given in `coala/community` room
+- Number of messages in coala other rooms
+- Make use of `corobo` on Gitter
+- Introduce `.coafile` in user own repository
+- Introduce `.coafile` in an organisation with 50 or more stars for its repositories
+- Introduce CI in the users own repository to run coala using Travis CI
+- Setup the CIs used by coala to your fork
+
+### Points
+
+Points are a simple tool to reward behavior, and they provide an excellent
+base upon which to build a gamified design. Now let's assign some points
+to each of the tracked activities mentioned above:
+
+But, as we are assigning points to the issues based on their labels which are
+initially labeled by the Gitmate-bot, we can not be sure all the time if the issue
+was labeled correctly or not. So, an issue should have `three` number of `Thumbs Up`
+reactions to qualify to be tracked in this system.
+
+Mainly it will be the developers' job to verify every new opening issues and give it
+the `Thumbs Up`, but in case if an issue seems to be labeled inappropriately by the bot,
+developers should request a maintainer to label it correctly and only then give `Thumbs
+Up` to the issue.
+
+Similarly in case of the pull requests if the mentioned issue in the commit is opened
+by newcomer and doesn't have `three` number of `Thumbs Up`, it will not qualify to
+be tracked in this system.
+
+- Create Issue
+  - 20 points for type/bug
+  - 20 points for bear proposal
+  - 10 points for difficulty newcomer
+  - 20 points for difficulty low
+  - 30 points for difficulty medium
+- Create Pull Requests
+  - No points will be given until all the CI on the pull request is passing
+  - If CI passes and there are no Gitmate errors
+    - 20 points for bug fixes
+    - 20 points for writing a bear
+    - 10 points for difficulty newcomer
```

Comment:
Do these points add up? For example, if I fix a difficulty/newcomer bug, do I get the 10 points for a difficulty/newcomer issue and 20 points for the bug fix? Maybe you could also add some points for adding a new feature?

Reply:
No, the points would not add up; Actually, I was assuming that we can not be sure about the difficulty level of the bug; That's why I assign a constant point to every issue labeled as a bug; But, yeah, when a bug has its difficulty level we will assign points only according to the difficulty level.